python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -613,160 +613,287 @@ def test_regular_args(ray_start_regular):
+        for i in range(3):
+            ref = compiled_dag.execute(x=2, y=3)
+            result = ray.get(ref)
+            assert result == [3, 2] * (i + 1)
 
+        with pytest.raises(
+            ValueError,
+            match=r"dag.execute\(\) or dag.execute_async\(\) must be called with kwarg",
+        ):
+            compiled_dag.execute()
 
-def test_kwargs_single_actor(ray_start_regular):
-    c = Collector.remote()
-    with InputNode() as i:
-        dag = c.collect_two.bind(i.y, i.x)
+        with pytest.raises(
+            ValueError,
+            match=r"dag.execute\(\) or dag.execute_async\(\) "
+            "must be called with kwarg `x`",
+        ):
+            compiled_dag.execute(y=3)
 
-    compiled_dag = dag.experimental_compile()
+        with pytest.raises(
+            ValueError,
+            match=r"dag.execute\(\) or dag.execute_async\(\) "
+            "must be called with kwarg `y`",
+        ):
+            compiled_dag.execute(x=3)
+
+        compiled_dag.teardown()
+
+    def test_kwargs_branch(self, ray_start_regular):
+        a = Actor.remote(0)
+        c = Collector.remote()
+        with InputNode() as i:
+            branch = a.inc.bind(i.x)
+            dag = c.collect_two.bind(i.y, branch)
+
+        compiled_dag = dag.experimental_compile()
 
-    for i in range(3):
         ref = compiled_dag.execute(x=2, y=3)
         result = ray.get(ref)
-        assert result == [3, 2] * (i + 1)
+        assert result == [3, 2]
 
-    with pytest.raises(
-        ValueError,
-        match=r"dag.execute\(\) or dag.execute_async\(\) must be called with kwarg",
-    ):
-        compiled_dag.execute()
+        compiled_dag.teardown()
 
-    with pytest.raises(
-        ValueError,
-        match=r"dag.execute\(\) or dag.execute_async\(\) must be called with kwarg `x`",
-    ):
-        compiled_dag.execute(y=3)
+    def test_multi_args_and_kwargs(self, ray_start_regular):
+        a1 = Actor.remote(0)
+        a2 = Actor.remote(0)
+        c = Collector.remote()
+        with InputNode() as i:
+            branch1 = a1.inc.bind(i[0])
+            branch2 = a2.inc.bind(i.y)
+            dag = c.collect_three.bind(branch2, i.z, branch1)
 
-    with pytest.raises(
-        ValueError,
-        match=r"dag.execute\(\) or dag.execute_async\(\) must be called with kwarg `y`",
-    ):
-        compiled_dag.execute(x=3)
+        compiled_dag = dag.experimental_compile()
 
-    compiled_dag.teardown()
+        ref = compiled_dag.execute(2, y=3, z=4)
+        result = ray.get(ref)
+        assert result == [3, 4, 2]
 
+        compiled_dag.teardown()
 
-def test_kwargs_branch(ray_start_regular):
-    a = Actor.remote(0)
-    c = Collector.remote()
-    with InputNode() as i:
-        branch = a.inc.bind(i.x)
-        dag = c.collect_two.bind(i.y, branch)
+    def test_mix_entire_input_and_args(self, ray_start_regular):
+        """
+        It is not allowed to consume both the entire input and a partial
```

Comment:
QQ: is it a behvior change or the same?

Reply:
No, it just hasn't been tested, and some ADAG logic is implemented based on this assumption. The check is in `preprocess`.

https://github.com/ray-project/ray/blob/35b453445b49ad286c08099953c000c3942acf6a/python/ray/dag/compiled_dag_node.py#L974-L978